algorithms/ar-msrt/typescript/merge_sort.ts

```diff
@@ -0,0 +1,30 @@
+function mergeSort(array: number[]) : number[]{
+    const merge = function (array1 : number[], array2 : number[]) : number[] {
+        if (array1 == [] || array1 == undefined){
+            return array2;
+        } else if (array2 == [] || array2 == undefined){
```

Comment:
Codacy found an issue: [Unexpected use of undefined.](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6334421)